### PR TITLE
TST: fix maybe_promote dt64tz case, 323 xfails

### PR DESCRIPTION
**Pull request by jbrockmendel**
about a third of our remaining xfailed tests

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -409,6 +410,14 @@ def maybe_promote(dtype, fill_value=np.nan):
     elif is_datetime64tz_dtype(dtype):
         if isna(fill_value):
             fill_value = NaT
+        elif not isinstance(fill_value, datetime):
+            dtype = np.dtype(np.object_)
+        elif fill_value.tzinfo is None:
+            dtype = np.dtype(np.object_)
+        elif not tz_compare(fill_value.tzinfo, dtype.tz):
+            # TODO: sure we want to cast here?
```
Comment (WillAyd) [inline pandas/core/dtypes/cast.py]: What is your apprehension with this comment?

Comment (jreback) [reply pandas/core/dtypes/cast.py]: actually you likley remove line 415 as None vs a tz will also not compare.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/dtypes/cast.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py]: > What is your apprehension with this comment?

For something like setitem, using a timestamp with a different tz is fine, so wouldn't cast to object.  so im a bit wary about doing this without having any real use cases that get here

Comment (jreback) [reply pandas/core/dtypes/cast.py]: k

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks.

**Pull request merged**